src/number_format.js

```diff
@@ -214,10 +219,13 @@ class NumberFormat extends React.Component {
   onKeyDown(e) {
     const el = e.target;
     const {selectionStart, selectionEnd, value} = el;
-    const {decimalPrecision} = this.props;
+    const {decimalPrecision, allowNegative} = this.props;
     const {key} = e;
     const numRegex = this.getNumberRegex(false, decimalPrecision !== false);
     //Handle backspace and delete against non numerical/decimal characters
+    if (key === '-' && allowNegative) {
+      this.setState({negative: !this.state.negative});
```

Comment:
Also, this will act as toggle, you will be able to press - at any cursor position and that will toggle the -, which shouldn't happen, user should be only allowed to type it before the number.